src/api/v1/db/oracledb/doughs-dao.js

```diff
@@ -0,0 +1,130 @@
+import config from 'config';
+import _ from 'lodash';
+
+import { serializeDoughs } from 'api/v1/serializers/doughs-serializer';
+import { openapi } from 'utils/load-openapi';
+import { getConnection } from './connection';
+
+const { endpointUri } = config.get('server');
+const doughsGetParameters = openapi.paths['/doughs'].get.parameters;
+
+/**
+ * @constant
+ * @type {object}
+ * @description an object mapping DoughRecipe property names to DOUGH table column names
+ */
+const doughColumnNames = {
+  id: 'ID',
+  name: 'NAME',
+  gramsFlour: 'GRAMS_FLOUR',
+  gramsWater: 'GRAMS_WATER',
+  flourType: 'FLOUR_TYPE',
+  waterTemp: 'WATER_TEMP',
+  gramsYeast: 'GRAMS_YEAST',
+  gramsSalt: 'GRAMS_SALT',
+  gramsSugar: 'GRAMS_SUGAR',
+  gramsOliveOil: 'GRAMS_OLIVE_OIL',
+  bulkFermentTime: 'BULK_FERMENT_TIME',
+  proofTime: 'PROOF_TIME',
+  specialInstructions: 'SPECIAL_INSTRUCTIONS',
+};
+
+/**
+ * @constant
+ * @type {string}
+ * @description a list of SQL aliases mapping DOUGH table column names to DoughRecipe properties,
+ *              generated from the DoughColumnNames object
+ */
+const doughColumnAliases = _.toPairs(doughColumnNames)
+  .map(([propertyName, columnName]) => `${columnName} AS "${propertyName}"`)
+  .join(', ');
+
+/**
+ *
+ * @param {string} filters
+ * @returns {string} The assembled query
+ */
+const doughsGetQuery = (filters) => `SELECT ${doughColumnAliases} FROM DOUGHS ${filters.length > 0 ? 'WHERE' : ''} ${filters}`;
+
+/**
+ *
+ * @param {string} filterName
+ * @returns {string} the filter name with 'filter' and brackets removed
+ */
+const filterToPropertyName = (filterName) => filterName.replace(/filter|\[|\]/g, '');
+
+/**
+ * @param {Array<string>} filters
+ * @returns {object} bindParams and Conditionals
+ */
+const processGetFilters = (filters) => {
+  const validFilters = doughsGetParameters.filter((filter) => filter.name in filters);
+  const conditionals = validFilters
+    .map((filter) => {
+      const propertyName = filterToPropertyName(filter.name);
+      return `${doughColumnNames[propertyName]} = :${propertyName}`;
+    })
+    .join(' AND ');
+
+  const bindParams = validFilters.reduce((params, filter) => {
+    params[filterToPropertyName(filter.name)] = filters[filter.name];
+    return params;
+  }, {});
+
+  return { bindParams, conditionals };
+};
```

Comment:
curious to hear feedback about code style here. I like using functional-style js programming but I think it can have readability issues if you're not careful.

Reply:
Fair enough. Personally, I like the fluent-interface style coding that you can do with the es6 methods, probably because it reminds me of ruby